pkg/prometheus/watcher.go

```diff
@@ -411,46 +448,99 @@ func (w *Watcher) createSvc(obj metav1.ObjectMeta, kind string, selector map[str
 	return annotations, nil
 }
 
-func (w *Watcher) formatAlertConfig(params []monitoringv1.Param) ([]byte, error) {
+func (w *Watcher) formatReceiver(amc *monitoringv1.AlertmanagerConfig, acfg *alertConfig) error {
+
+	switch amc.Spec.Type {
+	case "slack":
+		log.Debug("Formatting slack configuration")
+		err := w.formatSlackAlert(amc, acfg)
```

Comment:
Can we define formatter interface? Then have slack, pagerduty, email implement it. That way, adding new types will be easier. All this function does is call the format function of the right object.

Reply:
Done, created a formatter interface, slack and email implement them. Added test cases for this code.